src/nvim/lua/vim.lua

```diff
@@ -22,7 +22,7 @@ local function _update_package_paths()
       -- Find out path patterns. pathtrail should contain something like
       -- /?.so, /?/init.lua, /?.lua. This allows not to bother determining what
       -- correct suffixes are.
-      local pathtrail = s:match('[/\\][^/\\]*%?.*$')
+      local pathtrail = s:match('[/\\:][^/\\:]*%?.*$')
```

Comment:
Though adding colon was not a good idea, it looks like no moder systems support it.